File: matematica/numeros_primos/primalidade.cpp
// ============================================================================
// Primalidade
// O(√N)
// ============================================================================

/*

Dado um número inteiro não negativo N, o algoritmo verifica se N é primo.

*/

bool prime(const int n) {
    if (n < 2) {
        return false;
    }

    for (int i = 2; i * i <= n; i++) {
        if (n % i == 0) {
            return false;
        }
    }

    return true;
}

File: matematica/numeros_primos/primalidade.py
# Primalidade
# O(√N)

# Dado um número inteiro não negativo N, o algoritmo verifica se N é primo.

def primo(n: int) -> bool:
    if n < 2:
        return False

    i = 2
    while i * i <= n:
        if n % i == 0:
            return False
        i += 1

    return True
